data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/AsyncCloseable.java

```diff
@@ -61,7 +62,9 @@ static AutoCloseable toAutoCloseable(AsyncCloseable closeable) {
   static AsyncCloseable compose(AsyncCloseable... closeables) {
     return () -> CompositeFuture.all(
       Arrays.stream(closeables)
+        .filter(Objects::nonNull)
         .map(AsyncCloseable::close)
+        .filter(Objects::nonNull)
```

Comment:
should we reflect a bit what we actually do here w/ the streams APIs, on the javadoc of the `compose` method? It's currently pretty vague